pkg/kubecost/totals.go

```diff
@@ -18,22 +18,51 @@ import (
 // on-the-fly would be expensive; e.g. idle allocation; sharing coefficients
 // for idle or shared resources, etc.
 type AllocationTotals struct {
-	Start                time.Time `json:"end"`
-	End                  time.Time `json:"start"`
-	Cluster              string    `json:"cluster"`
-	Node                 string    `json:"node"`
-	Count                int       `json:"count"`
-	CPUCost              float64   `json:"cpuCost"`
-	GPUCost              float64   `json:"gpuCost"`
-	LoadBalancerCost     float64   `json:"loadBalancerCost"`
-	NetworkCost          float64   `json:"networkCost"`
-	PersistentVolumeCost float64   `json:"persistentVolumeCost"`
-	RAMCost              float64   `json:"ramCost"`
+	Start                          time.Time `json:"end"`
+	End                            time.Time `json:"start"`
+	Cluster                        string    `json:"cluster"`
+	Node                           string    `json:"node"`
+	Count                          int       `json:"count"`
+	CPUCost                        float64   `json:"cpuCost"`
+	CPUCostAdjustment              float64   `json:"cpuCostAdjustment"`
+	GPUCost                        float64   `json:"gpuCost"`
+	GPUCostAdjustment              float64   `json:"gpuCostAdjustment"`
+	LoadBalancerCost               float64   `json:"loadBalancerCost"`
+	LoadBalancerCostAdjustment     float64   `json:"loadBalancerCostAdjustment"`
+	NetworkCost                    float64   `json:"networkCost"`
+	NetworkCostAdjustment          float64   `json:"networkCostAdjustment"`
+	PersistentVolumeCost           float64   `json:"persistentVolumeCost"`
+	PersistentVolumeCostAdjustment float64   `json:"persistentVolumeCostAdjustment"`
+	RAMCost                        float64   `json:"ramCost"`
+	RAMCostAdjustment              float64   `json:"ramCostAdjustment"`
 }
 
-// TotalCost returns the sum of all costs
+// ClearAdjustments sets all adjustment fields to 0.0
+func (art *AllocationTotals) ClearAdjustments() {
+	art.CPUCostAdjustment = 0.0
+	art.GPUCostAdjustment = 0.0
+	art.RAMCostAdjustment = 0.0
+}
+
+// TotalCPUCost returns CPU cost with adjustment.
+func (art *AllocationTotals) TotalCPUCost() float64 {
+	return art.CPUCost + art.CPUCostAdjustment
+}
+
+// TotalGPUCost returns GPU cost with adjustment.
+func (art *AllocationTotals) TotalGPUCost() float64 {
+	return art.GPUCost + art.GPUCostAdjustment
+}
+
+// TotalRAMCost returns RAM cost with adjustment.
+func (art *AllocationTotals) TotalRAMCost() float64 {
+	return art.RAMCost + art.RAMCostAdjustment
+}
```

Comment:
What do you think about calling these `Adjusted...Cost()`?

Reply:
(same applies to `AssetTotals`)

Reply:
I agree that the naming with adjustments is weird. My goal is to communicate that *this is the one to use* so that people don't unwittingly use the unadjusted costs, thinking that "adjusted costs" aren't what they need—they just need to basic CPUCost. I don't know, maybe that's a bad justification.

In `allocation.go` we use `CPUTotalCost()` instead of `TotalCPUCost()` so I probably should've just stuck to convention. In general, I think this is messy no matter what, unfortunately.

I'm leaning toward keeping this the way it is, for the sake of ease. But I agree that it's not perfect, and that makes me a little sad. If we decide we like `AdjustedCPUCost()` then maybe we can take a pass and replace them all as a clean-up effort.